src/main/java/nextstep/subway/domain/PathFinder.java

```diff
@@ -0,0 +1,60 @@
+package nextstep.subway.domain;
+
+import lombok.extern.slf4j.Slf4j;
+import nextstep.subway.applicaion.constants.ErrorMessage;
+import nextstep.subway.applicaion.exception.PathFinderException;
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.DefaultWeightedEdge;
+import org.jgrapht.graph.WeightedMultigraph;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+
+@Slf4j
+public class PathFinder {
+
+    private List<Section> sections;
+    private WeightedMultigraph<Station, DefaultWeightedEdge> graph;
+
+    public PathFinder(List<Section> sections) {
+        this.sections = sections;
+        graph = new WeightedMultigraph<>(DefaultWeightedEdge.class);
+        setVertex(graph);
+        setEdge(graph);
+    }
+
+    private void setVertex(WeightedMultigraph<Station, DefaultWeightedEdge> graph) {
+        sections.stream()
+                .map(section -> Set.of(section.getUpStation(), section.getDownStation()))
+                .flatMap(Collection::stream)
+                .forEach(graph::addVertex);
+    }
+
+    private void setEdge(WeightedMultigraph<Station, DefaultWeightedEdge> graph) {
+        for (Section section : sections) {
+            DefaultWeightedEdge edge = graph.addEdge(section.getUpStation(), section.getDownStation());
+            graph.setEdgeWeight(edge, section.getDistance());
+        }
+    }
+
+    public GraphPath findPath(Station sourceStation, Station targetStation) {
+        DijkstraShortestPath dijkstraShortestPath = new DijkstraShortestPath(graph);
+        validatePath(sourceStation, targetStation);
+        GraphPath shortestPath = dijkstraShortestPath.getPath(sourceStation, targetStation);
+        if (shortestPath == null) throw new PathFinderException(ErrorMessage.NOT_FOUND_PATH);
```

Comment:
```suggestion
		if (Objects.isNull(shortestPath)) {
			throw new NotExistedPathException();
		}
```
`Objects.isNull` 을 활용해봐도 좋을 것 같아요. 그리고 이외 검증로직 처럼 메서드 분리해봐도 좋을 것 같네요!

Reply:
메서드 분리라는건 findPath를 분리하라는 말씀이신가요?
validatePath에 exception들을 몰아넣어놓는게 나은 방법인지는 잘 모르겠습니다

Reply:
shortestPath의 널 체크하는 검증 코드를 validPath 메서드와 같이 분리하여 통일성을 갖으면 좋을 것 같아 말씀드렸습니다!
저는 `valid-()` 와 같이 메서드 네이밍을 하면 예외 핸들링은 해당 메서드 내부에 두는게 더 나은 방법이라고 생각해요 ㅎㅎ